src/springfield/fields.py

```diff
@@ -390,12 +390,26 @@ def __init__(self, entity, *args, **kwargs):
                        Use 'self' to use the :class:`Entity` class that
                        this field is already bound to.
         """
-        self.type = entity
+        self._type = entity
         super(EntityField, self).__init__(*args, **kwargs)
 
+    @property
+    def type(self):
```

Comment:
Why is it a hack? It needs to be done after `__init__` - it originally was done in `init()`, but I don't know if that's safe/good/sufficient. Seemed better to do it the first time it's read.